internal/services/sarif/sarif.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2022 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sarif
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/ZupIT/horusec-devkit/pkg/entities/analysis"
+	"github.com/ZupIT/horusec-devkit/pkg/entities/vulnerability"
+	"github.com/ZupIT/horusec-devkit/pkg/enums/severities"
+
+	"github.com/ZupIT/horusec/cmd/app/version"
+)
+
+type Sarif struct {
+	analysiss *analysis.Analysis
+}
+
+func NewSarif(analysiss *analysis.Analysis) *Sarif {
+	return &Sarif{
+		analysiss: analysiss,
+	}
+}
+
+func (s *Sarif) ConvertVulnerabilityToSarif() (report Report) {
+	resultsByTool := make(map[string][]Result)
+	rulesByToolAndID := make(map[string]map[string]Rule)
+	artifactsByToolAndName := make(map[string]map[string]Artifact)
+
+	report.Runs = []ReportRun{}
+	s.populateReferenceMaps(&report, rulesByToolAndID, artifactsByToolAndName, resultsByTool)
+	s.buildReportRun(&report, rulesByToolAndID, artifactsByToolAndName, resultsByTool)
+
+	// SARIF output format version/schema
+	report.Version = "2.1.0"
+	report.SchemaURI = "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json"
+
+	return report
+}
+
+func (s *Sarif) populateReferenceMaps(
+	report *Report, rulesByToolAndID map[string]map[string]Rule,
+	artifactsByToolAndName map[string]map[string]Artifact, resultsByTool map[string][]Result) {
```

Comment:
This seems hard to read. Change to this pattern:
```go
func (s *Sarif) populateReferenceMaps(
	report *Report,
	rulesByToolAndID map[string]map[string]Rule,
	artifactsByToolAndName map[string]map[string]Artifact,
	resultsByTool map[string][]Result,
) {
```

Reply:
Unfortunately that change makes this too long for the linter. This is kind of related to the other issue of "should things have this many arguments" -- too many on one line and the line limit is exceeded, too many on different lines and it eats into the maximum number of lines per function.

I'm going to have to go back to the drawing board on how to put this together, I think.